rllib/env/multi_agent_env.py

```diff
@@ -555,6 +555,21 @@ def try_reset(self, env_id: Optional[EnvID] = None) -> Optional[MultiEnvDict]:
             ret[idx] = obs
         return ret
 
+    @override(BaseEnv)
+    def try_restart(self, env_id: Optional[EnvID] = None) -> None:
+        if isinstance(env_id, int):
+            env_id = [env_id]
+        if env_id is None:
+            env_id = list(range(len(self.envs)))
+        for idx in env_id:
+            # Recreate the sub-env.
+            self.envs[idx] = self.make_env(idx)
```

Comment:
make_env(idx) may potentially take a long time to come back.
just trying to double check, do you see any issues if the worker gets stuck here for a long time?

Reply:
Good point, but I don't think we can avoid this. When using lots of workers, it shouldn't be a problem, if one of them - very rarely - gets stuck for say 1 min or so ... ?
You could completely avoid this if you have `remote_worker_envs=True` (each sub-env is its own ray actor and stepping is parallel).

Reply:
Added another test with remote_worker_envs=True. Set init_time_s to 2.0 (so new created envs take some time to "come up"). Makes sure PGTrainer can learn even in these unstable conditions and that workers don't die b/c of any sub-env failures.